core/src/components/list/list.md.scss

```diff
@@ -23,12 +23,40 @@
   @include border-radius($list-inset-md-border-radius);
 }
 
-.list-md.list-inset ion-item:first-child {
+/**
+ * These selectors ensure the first item in the list
+ * has the correct radius.
+ * We need to consider the following scenarios:
+ 1. The only item in a list.
+ 2. The first item in a list as long as it is not the only item.
+ 3. The item in the first item-sliding in a list.
+ * Note that we do not select ion-item:first-of-type
+ * because that will cause the borders to disappear for
+ * when for items in item-sliding where the item is the last
+ * element in the item-sliding container.
```

Comment:
```suggestion
 * Note that we do not select ion-item:first-of-type
 * because that will cause the borders to disappear on
 * items in an item-sliding when the item is the first
 * element in the item-sliding container.
```